pkg/gpu/ebpf/c/gpu.c

```diff
@@ -0,0 +1,139 @@
+#define BPF_NO_PRESERVE_ACCESS_INDEX
+#define BPF_NO_GLOBAL_DATA
+
+#include "ktypes.h"
+#include "bpf_metadata.h"
+#include "bpf_helpers.h"
+#include "bpf_helpers_custom.h"
+#include "bpf_tracing.h"
+#include "compiler.h"
+#include "map-defs.h"
+
+#include "types.h"
+
+char __license[] SEC("license") = "GPL";
+
+BPF_PERF_EVENT_ARRAY_MAP(cuda_events, cuda_event_header_t);
+BPF_LRU_MAP(cuda_alloc_cache, __u64, cuda_alloc_request_args_t, 1024)
```

Comment:
Can we proactively clean this map in the `cudaFree` hook? Relying on the LRU eviction mechanism is not a good idea, especially as the number of cpus increase.

If this is not possible then you should pass the `BPF_F_NO_COMMON_LRU` flag to ensure that the freelists are per-cpu.

Reply:
Fixed, I forgot the cleanup.

Reply:
Since we are using LRU maps, I would still suggest to add
1) `BPF_F_NO_COMMON_LRU` flag if the LRU policy is not strictly required to evict the least recently used element.
2) If there is a possibility of map leaks then to register a userspace map cleaner.